flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/source/IcebergTableSource.java

```
@@ -53,7 +54,8 @@ public class IcebergTableSource
     implements ScanTableSource,
         SupportsProjectionPushDown,
         SupportsFilterPushDown,
-        SupportsLimitPushDown {
+        SupportsLimitPushDown,
+        SupportsSourceWatermark {
```

I think we have 2 feature in a single PR:
- CREATE TABLE LIKE
- Watermark support

Could we separate out these features to different PRs?
Could we write tests for both features?

---

These features were driven by mainly a use case, where an iceberg table is needed to be used in Flink window functions. This needs an incoming table to have MILLISECOND precision timestamp column and also watermark to be defined on source table. 

As iceberg only supports MICROSECOND timestamp columns, we need to have a table with computed columns and we can create these only in Flink Catalog. Iceberg catalog doesn't support creating tables with computed columns.

That said, i am happy to split them into 2 separate PR's. 
I have tests for CREATE TABLE LIKE.

As Watermark support is just making Source to implement interface and falling back to https://github.com/apache/iceberg/pull/9346 for core logic, i didn't have a test case. I can add a validation on if `watermark-column` is configured or not , so it can fail fast. And a test case around that.

---

Please separate out the features to 2 PR